src/views/shared/fiat-amount/fiat-amount.view.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+
+import { CurrencySymbol } from "src/utils/currencies";
+
+type FiatAmountProps = {
+  amount: number;
+  currency: string;
+};
+
+function FiatAmount({
+  amount,
+  currency,
+  className,
+}: FiatAmountProps & React.HTMLAttributes<HTMLDivElement>): JSX.Element {
+  if (
+    currency === "USD" ||
```

Comment:
Yes, I've realized this issue and no, `currency in CurrencySymbol` doesn't convince TS.

Indeed, I started writing a refactor and aborted but since you mentioned it, let me propose it here.

Reply:
The option I explored was keeping the const `CurrencySymbol` and extracting the type of the `code` to use it everywhere:

```diff
diff --git a/src/utils/currencies.ts b/src/utils/currencies.ts
index 0299c67..b09d8f3 100644
--- a/src/utils/currencies.ts
+++ b/src/utils/currencies.ts
@@ -4,6 +4,8 @@ import { getTokenAmountString } from "@hermeznetwork/hermezjs/src/utils";
 import { MAX_TOKEN_DECIMALS } from "src/constants";
 import { FiatExchangeRates, Token } from "src/domain/hermez";

+export type CurrencySymbolCode = typeof CurrencySymbol[keyof typeof CurrencySymbol]["code"];
+
 const CurrencySymbol = {
   USD: {
     symbol: "$",
@@ -25,7 +27,7 @@ const CurrencySymbol = {
     symbol: "£",
     code: "GBP",
   },
-};
+} as const;

 /**
  * Gets the string representation of a token amount with fixed decimals
diff --git a/src/views/shared/fees-table/fees-table.view.tsx b/src/views/shared/fees-table/fees-table.view.tsx
index e40773e..49395c9 100644
--- a/src/views/shared/fees-table/fees-table.view.tsx
+++ b/src/views/shared/fees-table/fees-table.view.tsx
@@ -9,6 +9,7 @@ import {
   getAmountInPreferredCurrency,
   getTokenAmountInPreferredCurrency,
   trimZeros,
+  CurrencySymbolCode
 } from "src/utils/currencies";
 // domain
 import { FiatExchangeRates, Token } from "src/domain/hermez";
@@ -18,7 +19,7 @@ interface FeesTableProps {
   l2Fee: number;
   estimatedWithdrawFee?: EstimatedWithdrawFee;
   token: Token;
-  preferredCurrency: string;
+  preferredCurrency: CurrencySymbolCode;
   fiatExchangeRates: FiatExchangeRates;
 }

diff --git a/src/views/shared/fiat-amount/fiat-amount.view.tsx b/src/views/shared/fiat-amount/fiat-amount.view.tsx
index fcfff5a..2b6a8f2 100644
--- a/src/views/shared/fiat-amount/fiat-amount.view.tsx
+++ b/src/views/shared/fiat-amount/fiat-amount.view.tsx
@@ -1,10 +1,10 @@
 import React from "react";

-import { CurrencySymbol } from "src/utils/currencies";
+import { CurrencySymbol, CurrencySymbolCode } from "src/utils/currencies";

 type FiatAmountProps = {
   amount: number;
-  currency: string;
+  currency: CurrencySymbolCode;
 };

 function FiatAmount({
@@ -12,22 +12,11 @@ function FiatAmount({
   currency,
   className,
 }: FiatAmountProps & React.HTMLAttributes<HTMLDivElement>): JSX.Element {
-  if (
-    currency === "USD" ||
-    currency === "EUR" ||
-    currency === "CNY" ||
-    currency === "JPY" ||
-    currency === "GBP"
-  ) {
-    return (
-      <span className={className}>
-        {CurrencySymbol[currency].symbol} {!isNaN(amount) ? amount.toFixed(2) : "--"}
-      </span>
-    );
-  } else {
-    console.error(`Currency symbol not available for the unsuported currency "${currency}"`);
-    return <span className={className}>{!isNaN(amount) ? amount.toFixed(2) : "--"}</span>;
-  }
+  return (
+    <span className={className}>
+      {CurrencySymbol[currency]} {!isNaN(amount) ? amount.toFixed(2) : "--"}
+    </span>
+  );
 }

 export default FiatAmount;
diff --git a/src/views/transactions/components/fee/fee.view.tsx b/src/views/transactions/components/fee/fee.view.tsx
index 5b3d6a1..5845e86 100644
--- a/src/views/transactions/components/fee/fee.view.tsx
+++ b/src/views/transactions/components/fee/fee.view.tsx
@@ -7,6 +7,7 @@ import {
   getAmountInPreferredCurrency,
   getFixedTokenAmount,
   getTokenAmountInPreferredCurrency,
+  CurrencySymbolCode,
 } from "src/utils/currencies";
 import { ReactComponent as AngleDownIcon } from "src/images/icons/angle-down.svg";
 import FeesTable from "src/views/shared/fees-table/fees-table.view";
@@ -22,7 +23,7 @@ interface FeeProps {
   fee: number;
   estimatedWithdrawFee?: EstimatedWithdrawFee;
   token: Token;
-  preferredCurrency: string;
+  preferredCurrency: CurrencySymbolCode;
   fiatExchangeRates: FiatExchangeRates;
   showInFiat: boolean;
   depositFee: EstimatedDepositFee;
diff --git a/src/views/transactions/components/transaction-amount-input/transaction-amount-input.view.tsx b/src/views/transactions/components/transaction-amount-input/transaction-amount-input.view.tsx
index f71426e..78ff215 100644
--- a/src/views/transactions/components/transaction-amount-input/transaction-amount-input.view.tsx
+++ b/src/views/transactions/components/transaction-amount-input/transaction-amount-input.view.tsx
@@ -5,7 +5,7 @@ import AmountInput from "src/views/shared/amount-input/amount-input.view";
 import useTransactionAmountInputStyles from "src/views/transactions/components/transaction-amount-input/transaction-amount-input.styles";
 import { ReactComponent as SwapIcon } from "src/images/icons/swap.svg";
 import { ReactComponent as ErrorIcon } from "src/images/icons/error.svg";
-import { getFixedTokenAmount } from "src/utils/currencies";
+import { getFixedTokenAmount, CurrencySymbolCode } from "src/utils/currencies";
 import FiatAmount from "src/views/shared/fiat-amount/fiat-amount.view";
 // domain
 import { HermezAccount } from "src/domain/hermez";
@@ -18,7 +18,7 @@ interface Amount {

 interface TransactionAmountInputProps {
   account: HermezAccount;
-  preferredCurrency: string;
+  preferredCurrency: CurrencySymbolCode;
   value: string;
   amount: Amount;
   showInFiat: boolean;
diff --git a/src/views/transactions/components/transaction-overview/transaction-overview.view.tsx b/src/views/transactions/components/transaction-overview/transaction-overview.view.tsx
index ba9aad2..2398d29 100644
--- a/src/views/transactions/components/transaction-overview/transaction-overview.view.tsx
+++ b/src/views/transactions/components/transaction-overview/transaction-overview.view.tsx
@@ -14,7 +14,11 @@ import Alert from "src/views/shared/alert/alert.view";
 import WithdrawInfoSidenav from "src/views/transactions/components/withdraw-info-sidenav/withdraw-info-sidenav.view";
 import TransactionInfoTable from "src/views/shared/transaction-info-table/transaction-info-table.view";
 import { AsyncTask } from "src/utils/types";
-import { getTokenAmountInPreferredCurrency, getFixedTokenAmount } from "src/utils/currencies";
+import {
+  getTokenAmountInPreferredCurrency,
+  getFixedTokenAmount,
+  CurrencySymbolCode,
+} from "src/utils/currencies";
 import { getRealFee } from "src/utils/fees";
 import {
   getPartiallyHiddenEthereumAddress,
@@ -34,7 +38,7 @@ import { EstimatedWithdrawFee } from "src/domain";
 type TransactionOverviewProps = {
   wallet: HermezWallet.HermezWallet;
   isTransactionBeingApproved: boolean;
-  preferredCurrency: string;
+  preferredCurrency: CurrencySymbolCode;
   fiatExchangeRates: FiatExchangeRates;
 } & (
   | {
```

Reply:
Back to your original comment, what about abstracting the manual check to a custom type guard in the currencies util?

Reply:
Ahá! I thought that TypeScript won't complain about that. I agree on move it to a type guard by the moment.

Reply:
Done.